### ISSUE TITLE
Icon Box Widget Accessibility - Icon Link Contains No Text

### ISSUE THREAD

**Issue by GermanPearls** [elementor/elementor#9565]
I've checked and don't believe this issue has been addressed.

The icon box widget is failing several Web Accessibility tests (ie: https://wave.webaim.org, https://www.deque.com/axe/) based on the lack of text on the icon link. An aria-hidden=true attribute was added to the icon via (https://github.com/elementor/elementor/issues/3269) which leaves an empty link with no text definition:

```
<div class="elementor-icon-box-wrapper">
  <div class="elementor-icon-box-icon">
    <a class="elementor-icon elementor-animation-grow" href="link_address">
      <i aria-hidden="true" class="fas fa-car"></i>
    </a>
  </div>
  <div class="elementor-icon-box-content">
    <h3 class="elementor-icon-box-title">
      <a href="link_address">Content title</a>
    </h3>
    <p class="elementor-icon-box-description">Content description</p>
  </div>
</div>
```

Perhaps the icon box title heading tag could get an id attribute and the icon's link tag could get an aria-labelledby attribute pointing to the heading.

```
<div class="elementor-icon-box-wrapper">
  <div class="elementor-icon-box-icon">
    <a aria-labelledby="icon-box-heading" class="elementor-icon elementor-animation-grow" href="link_address">
      <i aria-hidden="true" class="fas fa-car"></i>
    </a>
  </div>
  <div class="elementor-icon-box-content">
    <h3 id="icon-box-heading" class="elementor-icon-box-title">
      <a href="link_address">Content title</a>
    </h3>
    <p class="elementor-icon-box-description">Content description</p>
  </div>
</div>
```

**Comment by tiskraba** [comment]
I too have a problem with Accessibility of Icon Box.

In Icon Box both the Icon and the Header lead to the link provided in the settings, which is not correct.

**The correct sollution** would be for the icon not to have a href and only be an image with aria-hidden=true and alt="".

Explanation: duplicate links and duplicate focused navigation are also not good for accessibility. WAVE retruns them as warnings and it is something that should be avoided.

**Comment by Bovelett** [comment]
+1

**Comment by mpthornton** [comment]
Agreed. Thanks.

**Comment by formulaclick** [comment]
Same issue with "Icon Widget". ¿Any solution found for this? Maybe with a PHP filter we could manage to add custom attributes to anchor tag?

**Comment by ramiy** [comment]
@GermanPearls The new `id` attribute you suggested, should be unique to support multiple ion widgets on the page. Otherwise the `aria-labelledby` referral won't help.

**Comment by seshelby** [comment]
Hopefully this issue will be resolved sooner rather than later because it creates a major accessibility issue on many websites. In case anyone is interested I have added an option to the WP ADA Compliance  Check plugin to help with this issue. If a user inserts a custom attribute the plugin will convert it to an aria-label upon display. For example: data-screen-reader-text|Screen Reader Text

**Comment by morrisinc** [comment]
Any progress on getting this issue sorted?  Same errors on https://wave.webaim.org/ reports.

**Comment by seshelby** [comment]
This has actually be solved. If you click on the link settings for the icon you will see an attributes field. You can add aria-label|Screen reader text or title|screen reader text

**Comment by morrisinc** [comment]
Excellent - thanks that worked - thanks for your help.

**Comment by rami-elementor** [comment]
> This has actually been solved. If you click on the link settings for the icon you will see an attributes field. You can add `aria-label|Screen reader text` or `title|screen reader text`

Closing issue.

**Comment by Bovelett** [comment]
Thanks @rami-elementor :-)

**Comment by cmeurer** [comment]
Where would I find "the link settings for the icon"? I clicked on the edit pencil and don't see any place to "add aria-label|Screen reader text or title|screen reader text". Help!

**Comment by rami-elementor** [comment]
@cmeurer 

![image](https://github.com/elementor/elementor/assets/92088692/c82e82dc-50c8-43eb-9ae5-36363bda9352)

**Comment by cmeurer** [comment]
Thank you so much! Didn't know that was hiding there :)
